src/wrapper/clap-juce-wrapper.cpp

```diff
@@ -778,7 +796,55 @@ class ClapJuceWrapper : public clap::helpers::Plugin<
     bool noteNameGet(int index, clap_note_name *noteName) noexcept override
     {
         if (processorAsClapExtensions)
-            return processorAsClapExtensions->noteNameGet (index, noteName);
+            return processorAsClapExtensions->noteNameGet(index, noteName);
+        return false;
+    }
+
+    bool implementRemoteControls() const noexcept override
+    {
+        if (processorAsClapExtensions)
+            return processorAsClapExtensions->supportsRemoteControls();
+        return false;
+    }
+
+    uint32_t remoteControlsPageCount() noexcept override
+    {
+        if (processorAsClapExtensions)
+            return processorAsClapExtensions->remoteControlsPageCount();
+        return 0;
+    }
+
+    bool remoteControlsPageGet(uint32_t pageIndex,
+                               clap_remote_controls_page *page) noexcept override
+    {
+        if (processorAsClapExtensions)
+        {
+            juce::String sectionName{};
+            juce::String pageName{};
+            clap_id pageID{};
+            std::array<juce::AudioProcessorParameter *, CLAP_REMOTE_CONTROLS_COUNT> params{};
+
+            const auto result = processorAsClapExtensions->remoteControlsPageFill(
+                pageIndex, sectionName, pageID, pageName, params);
+
+            if (!result)
+                return false;
+
+            sectionName.copyToUTF8(page->section_name, CLAP_NAME_SIZE);
+            pageName.copyToUTF8(page->page_name, CLAP_NAME_SIZE);
+            page->page_id = pageID;
+            page->is_for_preset = false; // (Jatin) Not 100% sure what to do with this
+
+            for (size_t i = 0; i < CLAP_REMOTE_CONTROLS_COUNT; ++i)
+            {
+                if (params[i] == nullptr)
+                    page->param_ids[i] = CLAP_INVALID_ID;
```

Comment:
My only thought is should we return false here? Otherwise it looks great

Reply:
So I should probably document this a bit better, but the way Bitwig handles it, setting some of the `param_ids` to `CLAP_INVALID_ID` is basically how you tell the remote control page that you don't want to use all of the 8 parameter slots. So if I only wanted to show 3 parameters on the remote controls page, I need to set the other 5 slots to `CLAP_INVALID_ID`, otherwise Bitwig will try to fill those slots with other parameters.

Passing this up to the JUCE plugin layer, I figured the plugin could set whichever parameter slots it doesn't want to use to `nullptr` and then we could handle it here.

Reply:
OK that's perfect. Lets just document that decision in the header and merge!